packages/teleterm/src/ui/TopBar/Identity/IdentityList/IdentityListItem.tsx

```diff
@@ -32,33 +34,44 @@ export function IdentityListItem(props: IdentityListItemProps) {
     <ListItem
       onClick={props.onSelect}
       isActive={isActive}
-      onMouseEnter={() => setIsHovered(true)}
-      onMouseLeave={() => setIsHovered(false)}
+      ref={ref}
+      style={{ maxWidth: newWidth && newWidth + 'px' }}
+      onMouseEnter={() => {
+        // we set maxWidth to list item element, because otherwise it becomes wider on hover (new element is added to it)
+        setMaxWidth(ref.current.clientWidth);
```

Comment:
I strongly dislike solutions like this, but I think there's no way around it because the Popover element itself is not responsive and just uses `top` and `left` to position itself – I suppose the popover size wasn't meant to be changed while it is opened.

So I suspect there's no pure CSS solution to this. 😢

Reply:
I spend a lot of time trying to find CSS only solution, but I think it is not possible. As we don't want to set any restrictions on popover width there is no way to block changing it's width when we add a new element in row (which is logout button). Otherwise, popover width will jump on longer items hover.

I agree, JS solutions are problematic, however in this case it is very light and simple :)